### PR TITLE
deployment: use poetry, ES7 and invenio 3.3

### PR DESCRIPTION
**Pull request by rerowep**
* Upgrags to invenio 3.3.
* Uses poetry for python package handling.
* Changes elastic search version to 7.

Co-Authored-by: Peter Weber <peter.weber@rero.ch>

### CODE REVIEW

File: docker-services.yml [hunk 1]
```diff
@@ -82,8 +82,10 @@ services:
     image: docker.elastic.co/elasticsearch/elasticsearch-oss:6.6.0
```
Comment (blankoworld) [inline docker-services.yml:82]: Are you sure you change the elasticsearch version to 7? Or 6.6?

Comment (rerowep) [reply docker-services.yml:82]: Ups, I changed it to version 7.

File: scripts/__init__.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+# MA 02111-1307, USA.
+#
+# In applying this license, RERO does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""Poetry script utils."""
+
+import os
+# import subprocess
```
Comment (blankoworld) [inline scripts/__init__.py:28]: To be deleted, as next commented lines. Git is enough to remember code :wink:

File: scripts/__init__.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+import sys
+
+# def __getattr__(name):  # python 3.7+, otherwise define each script manually
+#     name = name.replace('_', '-')
+#     subprocess.run(
+#         ['python', '-u', '-m', name] + sys.argv[1:]
+#     )  # run whatever you like based on 'name'
+
```
Comment (blankoworld) [inline scripts/__init__.py:36]: To be deleted.

File: INSTALL.rst [hunk 4]
```diff
@@ -1,33 +1,32 @@
...
 
 Installation
 ============
 
-First you need to install
-`pipenv <https://docs.pipenv.org/install/#installing-pipenv>`_, it will handle
-the virtual environment creation for the project in order to sandbox our Python
-environment, as well as manage the dependency installation, among other things.
+You need to install `poetry`, it will handle the virtual environment creation for the project
+in order to sandbox our Python environment, as well as manage the dependency installation,
```
Comment (blankoworld) [inline INSTALL.rst:28]: I think you can change `our Python environment` by `the Python environment`.

File: test_idref.py [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+idref_uuids = open('/data/latest/idref_uuids.ids', 'r')
+log = open('/data/latest/idref_uuids.logs', 'w')
+for count, uuid in enumerate(idref_uuids):
+    uuid = uuid.strip()
+    idref_pid = '???'
+    try:
+        idref = IdrefRecord.get_record_by_id(uuid)
+        if idref:
+            idref_pid = idref.pid
+            print(count, uuid, idref_pid, end='\r', flush=True)
```
Comment (jma) [inline test_idref.py:17]: to be removed

File: test_idref.py [hunk 6]
```diff
@@ -0,0 +1,86 @@
+from invenio_db import db
```
Comment (jma) [inline test_idref.py:1]: copyright is missing but are you sure that this file should be a part of this commit? If yes, it should be either a   cli script or a test function in the `tests` directory.

File: scripts/bootstrap [hunk 7]
```diff
@@ -92,41 +92,46 @@ do
...
 fi
 
-# install assets utils
-virtualenv_path=`pipenv --venv`
-info_msg "Install npm assets utils in: ${virtualenv_path}"
-npm i npm@latest -g --prefix `pipenv --venv` && pipenv run npm install --prefix `pipenv --venv` --silent -g node-sass@4.9.0 clean-css@3.4.19 uglify-js@2.7.3 requirejs@2.2.0 @angular/cli@7.0.4
+# # install assets utils
+# virtualenv_path=`poetry env info -p`
+# info_msg "Install npm assets utils in: ${virtualenv_path}"
+# npm i npm@latest -g --prefix "${virtualenv_path}" && npm install --prefix "${virtualenv_path}" --silent -g node-sass@4.9.0 clean-css@3.4.19 uglify-js@2.7.3 requirejs@2.2.0 @angular/cli@7.0.4 yarn
```
[the remark this answers is not in the document]
Comment (rerowep) [reply scripts/bootstrap:128]: Yes but not the same way as rero-ils

File: pyproject.toml [hunk 8]
```diff
@@ -0,0 +1,100 @@
+[tool.poetry]
+name = "rero-mef"
+version = "0.3.0"
+description = "Invenio digital library framework MEF."
+authors = ["RERO <software@rero.ch>"]
+license = "GNU Affero General Public License v3.0"
+
+[tool.poetry.dependencies]
+python = ">= 3.6, < 3.8"
```
[the remark this answers is not in the document]
Comment (rerowep) [reply pyproject.toml:9]: ok

File: rero_mef/authorities/bnf/mappings/v7/bnf/bnf-person-v0.0.1.json [hunk 9]
```diff
@@ -0,0 +1,55 @@
+{
+  "settings": {
+    "number_of_shards": 1,
+    "number_of_replicas": 0,
```
[the remark this answers is not in the document]
Comment (rerowep) [reply rero_mef/authorities/bnf/mappings/v7/bnf/bnf-person-v0.0.1.json:4]: Ok will try with 1

File: rero_mef/receivers.py [hunk 10]
```diff
@@ -35,7 +35,7 @@ def extend_mef_record(
     **kwargs
 ):
     """Extend MEF record with list of sources."""
-    if doc_type.startswith('mef'):
+    if index.startswith('mef'):
```
[the remark this answers is not in the document]
Comment (rerowep) [reply rero_mef/receivers.py:38]: I thing this changed from version 3.1 to 3.2. It is a corrections to work again in version 3.2.

File: test_idref.py (hunk 6 above)
Comment (rerowep) [reply test_idref.py:1]: Where does that come from?

File: test_idref.py (hunk 5 above)
Comment (rerowep) [reply test_idref.py:17]: Where does that come from?

File: pyproject.toml [hunk 11]
```diff
@@ -0,0 +1,100 @@
+[tool.poetry]
+name = "rero-mef"
+version = "0.3.0"
+description = "Invenio digital library framework MEF."
```
[the remark this answers is not in the document]
Comment (rerowep) [reply pyproject.toml:4]: ok

File: rero_mef/authorities/bnf/mappings/v7/bnf/bnf-person-v0.0.1.json (hunk 9 above)
[the remark this answers is not in the document]
Comment (rerowep) [reply rero_mef/authorities/bnf/mappings/v7/bnf/bnf-person-v0.0.1.json:4]: We have the same settings in rero-ils and rero-ebooks.

File: rero_mef/utils.py [hunk 12]
```diff
@@ -200,7 +200,7 @@ def oai_process_records_from_dates(name, sickle, oai_item_iterator,
     last_run = None
     url, metadata_prefix, last_run, setspecs = get_info_by_oai_name(name)
 
-    request = sickle(url, iterator=oai_item_iterator)
+    request = sickle(url, iterator=oai_item_iterator, max_retries=10)
```
Comment (jma) [inline rero_mef/utils.py:203]: can you put a small comment to explain why?

File: test_idref.py (hunk 6 above)
Comment (jma) [inline test_idref.py:1]: licence is missing, can you add a script to check?

Comment (jma) [reply test_idref.py:1]: This file looks like a debug file is it really a part of this commit?

Comment (rerowep) [reply test_idref.py]: deleted

File: rero_mef/utils.py (hunk 12 above)
Comment (rerowep) [reply rero_mef/utils.py]: corrected differently

**Pull request merged**